### DISCUSSION TITLE
Generate ISO file from edited installation

### DISCUSSION THREAD

**Discussion by pbvafer** [cobbler/cobbler#3209] [Version specific questions]
@SchoolGuy I have the doubt if once we have installed with Cobbler an operating system automatically, we can after installing drivers, packages, and more software, generate an ISO file that accepts Cobbler and import it to the repository. I have tested with Clonezilla, mkisofs, dd and when generating an ISO file the structure is not identical to a virgin one and Cobbler does not validate it.

**Comment by SchoolGuy** [comment]
Do I understand correctly that the problem is the following:

1. Take an OS ISO and unpack it
2. Modify it to suit the environment
3. Pack it as an ISO again
4. Use `cobbler import` to add the modified distro to Cobbler

The problem seems to be step four, correct?

**Reply by pbvafer** [reply]
Yes, Cobbler Fails in this step

**Reply by pbvafer** [reply]
![ImportFailCobblerClon](https://user-images.githubusercontent.com/79220152/180751968-0cccfabf-ad76-41fe-a97a-b18d13e546ef.png)

**Reply by SchoolGuy** [reply]
What you can do is you can use the manual workflow and use a `cobbler distro add`, along with `cobbler profile add` and `cobbler repo add` if that doesn't work automatically. If you insist on using `cobbler import` and the distro is derived that much from the upstream distro that it isn't recognized anymore you should be able to adjust ´signatures.json` in a way that suits you. We have a tool called [`libcobblersignatures`](https://github.com/cobbler/libcobblersignatures) that is in beta stadium that should be able to simplify this process in a great deal.

**Reply by pbvafer** [reply]
@SchoolGuy Hi, do you know of a tool to package the entire runtime environment in an ISO file?

**Reply by SchoolGuy** [reply]
@pbvafer What do you consider the entire runtime? So everything including the OS?

**Reply by pbvafer** [reply]
Yes, drivers, applications, full O.S.

**Reply by SchoolGuy** [reply]
Well if you run a VM then it is just a matter of backing that up. If you have bare-metal it is a bit more complicated. There are various backup tools available. I am not aware of the limitations you have in your environment but the low-level way would be to `dd` the physical drive to another drive. However the system obviously needs to be shut down, which is not very good for most setups. I never used it but I have heard of [Clonezilla](https://clonezilla.org/). If you need something enterprisy then have a look at [Veeam](https://www.veeam.com/). But there is a ton of other options available, all with their own implications.

**Reply by pbvafer** [reply]
Ok, thank you very much, it's a bare-metal installation, I've already used Clonezilla with the Cobbler verification issue I mentioned above, I'll give it a try.

**Reply by pbvafer** [reply]
@SchoolGuy Hi! I keep having problems importing modified ISO files into Cobbler, do you know of anyone who has done this who can help me? It's important to me and I'm not making any progress. I have used Clonezilla as you told me and also other ways and I have not been able to make progress in any of them. Thank you in advance.

**Reply by SchoolGuy** [reply]
I sadly don't know anyone I could ping. I will try myself but in theory it should work as long as the signature is matching the distro of the normal OS version you are using as a base. In case you are modifying the system to an extent where the signatures from `signatures.json` doesn't match anymore, then you need to add an entry to the JSON to match your custom OS.
